packages/compass-components/src/components/icon-badge.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { spacing } from '@leafygreen-ui/tokens';
+import { css } from '@leafygreen-ui/emotion';
+import { Variant } from '@leafygreen-ui/badge';
+
+import { Badge, Icon } from './leafygreen';
+
+const iconButtonStyles = css({
+  padding: 0,
+  background: 'transparent',
+  border: 'none',
+  lineHeight: 0,
+  cursor: 'pointer',
+});
+
+const contentStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[1],
+});
+
+type IconBadgeProps = {
+  text: string;
+  icon: string;
+  dataTestId?: string;
+  onClick?: () => void;
+  variant?: Variant;
+};
+
+function IconBadge({
```

Comment:
If this is a shared, reusable component, should we use it in other places where IconBadge is used? For example in collections list?

<img width="202" alt="image" src="https://user-images.githubusercontent.com/5036933/182322285-747f0dd5-dcdb-468e-9db6-a25a7679e4ca.png">

Or is this very specific `BadgeWithIconButton` or something?

Reply:
I ended up removing this component to keep everything simple